devito/passes/iet/languages/CXX.py

```diff
@@ -0,0 +1,63 @@
+from devito.ir import Call, UsingNamespace
+from devito.passes.iet.langbase import LangBB
+
+__all__ = ['CXXBB']
+
+
+std_arith = """
+#include <complex>
+
+template<typename _Tp, typename _Ti>
+std::complex<_Tp> operator * (const _Ti & a, const std::complex<_Tp> & b){
+  return std::complex<_Tp>(b.real() * a, b.imag() * a);
+}
+
+template<typename _Tp, typename _Ti>
+std::complex<_Tp> operator * (const std::complex<_Tp> & b, const _Ti & a){
+  return std::complex<_Tp>(b.real() * a, b.imag() * a);
+}
+
+template<typename _Tp, typename _Ti>
+std::complex<_Tp> operator / (const _Ti & a, const std::complex<_Tp> & b){
+  _Tp denom = b.real() * b.real () + b.imag() * b.imag()
+  return std::complex<_Tp>(b.real() * a / denom, - b.imag() * a / denom);
+}
+
+template<typename _Tp, typename _Ti>
+std::complex<_Tp> operator / (const std::complex<_Tp> & b, const _Ti & a){
+  return std::complex<_Tp>(b.real() / a, b.imag() / a);
+}
+
+template<typename _Tp, typename _Ti>
+std::complex<_Tp> operator + (const _Ti & a, const std::complex<_Tp> & b){
+  return std::complex<_Tp>(b.real() + a, b.imag());
+}
+
+template<typename _Tp, typename _Ti>
+std::complex<_Tp> operator + (const std::complex<_Tp> & b, const _Ti & a){
+  return std::complex<_Tp>(b.real() + a, b.imag());
+}
+
+"""
+
+
+class CXXBB(LangBB):
+
+    mapper = {
+        'header-memcpy': 'string.h',
+        'header-algorithm': 'algorithm',
```

Comment:
I'd call this `header-math`

Reply:
The name of the header is algorithm it contains more than just math stuffs in it, the math header is `math.h`

Reply:
it doesn't matter, that's our own semantics. The mapper gives us an entry for the "header that contains the mathematical definitions". This way, all the languages could reuse the same key `'header-math'` , while `'header-algorithm'` would only be a C++ thing

Reply:
ping

Reply:
Sure but that's confusing cuz everyone would expect `header-math` to be `math.h` .  Maybe `header-math-ago` ?